networkx/algorithms/distance_measures.py

```diff
@@ -1029,3 +1032,175 @@ def kemeny_constant(G, *, weight=None):
 
     # Compute the Kemeny constant
     return float(np.sum(1 / (1 - eig[:-1])))
+
+
+def _four_sweep_start_vertex(G, choice):
+    """Selects a start vertex for the four-sweep BFS process.
+
+    Parameters:
+    G : NetworkX graph
+        The input graph.
+    choice : str
+        The selection strategy for the start vertex. Options:
+        - "random": Randomly selects a node.
+        - "degree": Selects the node with the highest degree.
+
+    Returns:
+    node
+        The selected start vertex.
+    """
+    if choice == "random":
+        start_vertex = random.choice(list(G.nodes()))
+    elif choice == "degree":
+        start_vertex = max(G.nodes(), key=lambda x: G.degree(x))
+    else:
+        raise ValueError("Invalid choice of start vertex")
```

Comment:
This could be extended to allow the user to provide a function with input `G` that returns the starting node.
But that also can be done after this PR is merged. Also the syntax `key=G.degree` might be better than the `lambda`. Again, not a blocker.